### ISSUE TITLE
API docs server does not work with chrome 73

### ISSUE THREAD

**Issue by wutzx** [caicloud/nirvana#267]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

I got empty webpage when i access api docs server from chrome 73, and api docs server report below log:

ERROR 0319-18:34:14.402+08 builder.go:318 | strconv.ParseFloat: parsing "b3": invalid syntax

**What you expected to happen**: 

I can get right webpage about api docs.

**How to reproduce it (as minimally and precisely as possible)**:

Use chrome 73 to access api docs server which start by `nirvana api`.

**Anything else we need to know?**:

Chrome 73 request with specify HTTP header `Accept: v=b3`, details:

`Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3`

**Comment by kdada** [comment]
The issue is from `service/filter.go#129` `parseAcceptTypes`.  It seems that the function has some bugs. I'm digging into it.

**Comment by kdada** [comment]
I upgrade my chrome to 73.0.3683.75. Everything is ok. Do you use the latest version of nirvana?

**Comment by wutzx** [comment]
Yes, i am using laster version of nirvana.

**Comment by kdada** [comment]
@wutzx Could you provide a minimal project for this case?

**Comment by wutzx** [comment]
https://github.com/wutzx/myproject

**Comment by kdada** [comment]
@wutzx I tested your project and there is no problem with `nirvana api`.

**Comment by wutzx** [comment]
So this issue may be related to my chrome environment, but i do not known why request with `v=b3`

**Comment by kdada** [comment]
@wutzx In my env, the `Accept` is also `text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3`. But nirvana handles this cases correctly.

**Comment by kdada** [comment]
You can fetch the latest nirvana from master branch and reinstall the nirvana command.

**Comment by wutzx** [comment]
Ok, i will be to trying later.

Wei Guo <notifications@github.com>于2019年3月20日 周三下午12:01写道：

**Comment by wutzx** [comment]
I am reinstall nirvana, then it can work correctly. May be nirvana to provide a flag `--version` to quick known whether is the lastest version.

**Comment by kdada** [comment]
> I am reinstall nirvana, then it can work correctly. May be nirvana to provide a flag `--version` to quick known whether is the lastest version.

It's a good idea.

**Comment by hanxueluo** [comment]
I have reproduced this issue in Chrome 73.0.3683.86. Server replies an empty body, but status code == 200. After debug, I find error is from following code line.
```
func parseAcceptTypes(v string) ([]string, error) {
	types := []string{}
	factors := []float64{}
	strs := strings.Split(v, ",")
	for _, str := range strs {
		str := strings.Trim(str, " ")
		tf := strings.Split(str, ";")
		types = append(types, tf[0])
		factor := 1.0
		if len(tf) == 2 {
			qp := strings.Split(tf[1], "=")
			q, err := strconv.ParseFloat(qp[1], 32)
			if err != nil {
				return nil, err // <== strconv.ParseFloat raises an error, because qp[1] is "b3"
			}
			factor = q
		}
		factors = append(factors, factor)
	}
```

**Comment by kdada** [comment]
@hanxueluo please fetch the latest commit from the master.

**Comment by kdada** [comment]
I'v published `v0.2.1`. I'm gonna close this issue. If there is any problem has not solved, feel free to reopen this issue.
